### PR TITLE
maestral-gui: 1.4.6 -> 1.4.8

### PR DESCRIPTION
**Pull request by SFrijters**
###### Motivation for this change

Upstream update, including update for dependency desktop-notifier.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135258` at 5315bb16 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>maestral (python39Packages.maestral)</li>
    <li>maestral-gui</li>
    <li>python38Packages.desktop-notifier</li>
    <li>python38Packages.maestral</li>
    <li>python39Packages.desktop-notifier</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/maestral/default.nix:51:0:
```
   |
51 |     description = "Open-source Dropbox client for macOS and Linux";
   | ^
```
</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> warning: python-include-tests
> Consider adding a checkPhase for tests, or if not feasible, pythonImportsCheck.

Can you add this?

**Comment by SFrijters** [comment]
Done, I think.

File: pkgs/development/python-modules/maestral/default.nix [hunk 1]
```diff
@@ -44,8 +45,26 @@ buildPythonPackage rec {
...
+  disabledTests = [
+    # Tries to look at /usr
+    "test_filestatus"
+    # Requires systemd
+    "test_autostart"
+    # Network access
+    "test_check_for_updates"
+    # Tries to look at /usr/local/share
+    "test_path_exists_case_insensitive"
+    # Tried to look at /usr/local/share
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/maestral/default.nix:63]:
```suggestion
```

File: pkgs/development/python-modules/maestral/default.nix [hunk 2]
```diff
@@ -44,8 +45,26 @@ buildPythonPackage rec {
     "--prefix" "PYTHONPATH" ":" "$out/lib/${python.libPrefix}/site-packages"
   ];
 
-  # no tests
-  doCheck = false;
+  checkInputs = [
+    pytestCheckHook
+    pytest-benchmark
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/maestral/default.nix:50]: We don't want to benchmark. Can we disable this feature in setup.py/etup,cfg?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/maestral/default.nix (hunk 2 above)
Comment (SFrijters) [reply pkgs/development/python-modules/maestral/default.nix]: I've disabled the single `"test_performance"` benchmark, is that good enough? I don't know how to do it from setup.cfg, would it need to be patched?

File: pkgs/development/python-modules/maestral/default.nix (hunk 1 above)
Comment (SFrijters) [reply pkgs/development/python-modules/maestral/default.nix]: Reshuffled the tests and comments a bit.

File: pkgs/development/python-modules/maestral/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/maestral/default.nix]: > I've disabled the single "test_performance" benchmark, is that good enough?

That should be fine.

> don't know how to do it from setup.cfg, would it need to be patched?

Yeah, we would most likely do it in postPatch with substituteInPlace.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 135258` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>maestral (python39Packages.maestral)</li>
<li>maestral-gui</li>
<li>python38Packages.desktop-notifier</li>
<li>python38Packages.maestral</li>
<li>python39Packages.desktop-notifier</li>
</ul>
</details>

**Pull request merged**